File: Assignment 1/C++/q2.cpp
#include<iostream>
using namespace std;
int main(){
        double n;
        cout<<"Choose a number of your choice to compute factorial:\n";
        cin>>n;
        double x;
        double m;
        x=n;
        m=n;
        if(n==0){
            cout<<"The value of 0! is 1";
                }
        else if(n>0){
            if(n==1){
                cout<<"The value of 1! is 1";
                    }
            else{
                while(n>1){
                x=x*(n-1);
                n--;
                   }
                cout<<"The value of "<<m<<"! is "<<x;
                 }
               }
        else{
            cout<<"The number you entered is negative!,please enter any positive number(zero included)";
            }
return 0;
}

File: Assignment 1/Python/q2.py
n=int(input('Choose a number of your choice to compute factorial:\n'))
x=n
m=n
if n==0:
    print('The value of 0! is 1') 
else:
    if n>0:
        if n==1:
            print('The value of 1! is 1')
        else: 
            while n>1:
                x=x*(n-1)
                n=n-1
            print('The value of ', m,'! is',x)
    else: 
        print('The number you entered is negative!,please enter any positive number(zero included)')
       
        